src/map-view.ios.ts

```diff
@@ -275,20 +275,79 @@ class MapViewDelegateImpl extends NSObject implements GMSMapViewDelegate {
     }
 }
 
+@NativeClass()
+class MapVCDelegateImpl extends NSObject implements CLLocationManagerDelegate
+{
+    public static ObjCProtocols = [CLLocationManagerDelegate];
+
+    private _owner: WeakRef<MapView>;
+    public lm: CLLocationManager = new CLLocationManager();
+
+    public static initWithOwner(owner: WeakRef<MapView>): MapVCDelegateImpl {
+        let handler = <MapVCDelegateImpl>MapVCDelegateImpl.new();
+        handler._owner = owner;
+        return handler;
+    }
+
+    public locationManagerDidChangeAuthorizationStatus(manager: CLLocationManager, status: CLAuthorizationStatus)
+    {
+        let owner = this._owner.get();
+        switch (status)
+        {
+        case CLAuthorizationStatus.kCLAuthorizationStatusAuthorizedAlways:
+            console.log("Location AuthorizedAlways")
+            owner.myLocationEnabled = true
+            this.lm.startUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusAuthorizedWhenInUse:
+            console.log("Location AuthorizedWhenInUse")
+            owner.myLocationEnabled = true
+            this.lm.startUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusDenied:
+            console.log("Location Denied")
+            owner.myLocationEnabled = false
+            this.lm.stopUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusNotDetermined:
+            console.log("Location NotDetermined")
+            owner.myLocationEnabled = false
+            this.lm.stopUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusRestricted:
+            console.log("Location Restricted")
+            owner.myLocationEnabled = false
+            this.lm.stopUpdatingLocation()
+        }
+    }
+
+    public locationManagerDidUpdateLocations(manager: CLLocationManager, locations: NSArray<CLLocation> | CLLocation[])
+    {
+        let owner = this._owner.get();
+        console.log(locations)
+        /*if (locations.length > 0)
+        {
+            owner.gMap.camera = GMSCameraPosition.cameraWithTargetZoom(locations.coordinate, 10.0)
+            owner.settings.myLocationButton = true
+        }*/
+    }
+}
 
 export class MapView extends MapViewBase {
 
     protected _markers: Array<Marker> = new Array<Marker>();
 
-    private _delegate: MapViewDelegateImpl;
+    public _delegate: MapViewDelegateImpl;
     private _indoorDelegate:IndoorDisplayDelegateImpl;
+    //private _mapVCDelegate:MapVCDelegateImpl;
 
     constructor() {
         super();
 
         this.nativeView = GMSMapView.mapWithFrameCamera(CGRectZero, this._createCameraPosition());
         this._delegate = MapViewDelegateImpl.initWithOwner(new WeakRef(this));
```

Comment:
WeakRef class seem a quite busy one (updated often by NS), this initialization could be another source of problems

Reply:
I think we dont have the choice in order to implement IOS delegation!